.github/workflows/set-metadata.yml

```diff
@@ -0,0 +1,385 @@
+          echo "Tagged version converted to PEP 440 standard: $pepified_version"
+          echo "version=$pepified_version" >> $GITHUB_OUTPUT
+
+      - name: Construct release tag
+        if: needs.set-metadata.outputs.build-type == 'release'
+        id: set-release-version
+        run: |
+          RELEASE_TAG=""
+
+          # If tag not given by user, deduce from commit titles since last tag:
+          if test -z "${{ inputs.optional-release-tag }}"; then
+            echo "Autoincrementing version based on previous commit titles since last tag"
+
+            # Search previous commits for MAJOR/MINOR text tokens:
+            IS_MAJOR_BUMP=false
+            IS_MINOR_BUMP=false
+            echo "Checking commit titles since last tag: '${{ steps.previous-tag.outputs.tag }}'..."
+            COMMITS=$(git log --pretty=format:%s ${{ steps.previous-tag.outputs.tag }}..@)
+            while read commit
+            do
+              if [[ $commit == *"MAJOR"* ]]; then
+                echo -e "\tMajor: title='$commit'"
+                IS_MAJOR_BUMP=true
+              elif [[ $commit == *"MINOR"* ]]; then
+                echo -e "\tMinor: title='$commit'"
+                IS_MINOR_BUMP=true
+              else
+                echo -e "\tPatch: title='$commit'"
+              fi
+            done <<< "$COMMITS"
+
+            # Set version according to what was found in commit titles:
+            if [ "$IS_MAJOR_BUMP" = true ]; then
+              echo "Bumping major version"
+              RELEASE_TAG=${{ steps.next-semvers.outputs.major }}
+            elif [ "$IS_MINOR_BUMP" = true ]; then
+              echo "Bumping minor version"
+              RELEASE_TAG=${{ steps.next-semvers.outputs.minor }}
+            else
+              # If no major/minor found, treat as patch:
+              echo "Bumping patch version"
+              RELEASE_TAG=${{ steps.next-semvers.outputs.patch }}
+            fi
+          else
+            echo "Using user provided release tag"
+            RELEASE_TAG=${{ inputs.optional-release-tag }}
+          fi
+
+          # Check if valid semver:
+          regex='^([0-9]+\.){2}(\*|[0-9]+)(-.*)?$'
+          if [[ ! "$RELEASE_TAG" =~ $regex ]]; then
+            echo "❌ - Release tag is not a valid semver: $RELEASE_TAG"
+            exit 1
+          fi
+          echo "✔ - Release tag is a valid semver"
+
+          # Check if tag already exists:
+          if git rev-parse "$RELEASE_TAG" >/dev/null 2>&1; then
+            echo "❌ - Release tag already exists: $RELEASE_TAG"
+            exit 1
+          fi
+          echo "✔ - Release tag does not exist"
+
+          echo "Release tag: $RELEASE_TAG"
+          echo "version=$(echo $RELEASE_TAG)" >> $GITHUB_OUTPUT
+    
+      - name: choose-version
+        id: choose-version
+        run: |
+          if [[ -n "${{ steps.pepify.outputs.version }}" ]]; then
+            echo "Setting pepified PR/branch version"
```

Comment:
This version comes out of the pepify step but it actually might not be pepified (in the case of non-python repos). If you don't want to rework that, I would just change the echo statement here and remove "pepified" then add displaying the version you are actually setting. Same for the else with adding the details about which version is being set.